godot-core/src/builtin/transform2d.rs

```diff
@@ -320,6 +320,25 @@ impl Mul<real> for Transform2D {
     }
 }
 
+impl Mul<Rect2> for Transform2D {
+    type Output = Rect2;
+
+    /// Transforms each coordinate in `rhs.position` and `rhs.end()` individually by this transform, then
+    /// creates a `Rect2` containing all of them.
+    fn mul(self, rhs: Rect2) -> Self::Output {
+        // https://web.archive.org/web/20220317024830/https://dev.theomader.com/transform-bounding-boxes/
+        let xa = self.a * rhs.position.x;
+        let xb = self.a * rhs.end().x;
+
+        let ya = self.b * rhs.position.y;
+        let yb = self.b * rhs.end().y;
+
+        let position = Vector2::min(xa, xb) + Vector2::min(ya, yb) + self.origin;
+        let end = Vector2::max(xa, xb) + Vector2::max(ya, yb) + self.origin;
+        Rect2::new(position, end - position)
```

Comment:
What do you think, would begin/end constructors make sense?
```rs
Rect2::from_corners(position, end)
Aabb::from_corners(position, end)
```

Reply:
probably yeah tbh

Reply:
I would probably stay with `position`/`end`, because that's also the name of the corresponding fields or getters/setters.